src/librustc_data_structures/small_c_str.rs

```diff
@@ -11,69 +11,61 @@
 use std::ffi;
 use std::ops::Deref;
 
-const SIZE: usize = 38;
+use smallvec::SmallVec;
+
+const SIZE: usize = 36;
```

Comment:
Any reason for the change?

Reply:
Yes, `Array` isn't implemented for `[T; 38]`, and due to the union optimization, the on-stack version can take more content per byte.